components/feral-connectd/websocket.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// WSClient handles WebSocket connection to relay server
+type WSClient struct {
+	config *Config
```

Comment:
Should store the locationID and topicID, the whole global configuration could be a mess.